### PR TITLE
[Neo Core Store] Rename various snapshots.

### PR DESCRIPTION
**Pull request by Jim8y**
# Description

The name of snapshot has being misused almost everywhere in the core, snapshotcache is different from snapshot, and clonedcache is also not a snapshot. Misuse the terminology of snapshot confuses developers. 

Fixes # (issue)

## Type of change

- [ ] Optimization (the change is only an optimization)
- [x] Style (the change is only a code style for better maintenance or standard purpose)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

**Test Configuration**:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/Neo/NeoSystem.cs [hunk 1]
```diff
@@ -275,12 +275,24 @@ public void SuspendNodeStartup()
...
             return new SnapshotCache(store.GetSnapshot());
         }
 
+        /// <summary>
+        /// Gets a snapshot of the blockchain storage with an execution cache.
+        /// With the snapshot, we have the latest state of the blockchain, with the cache,
+        /// we can run transactions in a sandboxed environment.
+        /// </summary>
+        /// <returns>An instance of <see cref="SnapshotCache"/></returns>
+        public SnapshotCache GetSnapshotCache()
```
Comment (cschuchardt88) [inline src/Neo/NeoSystem.cs:291]: This should be called `CreateSnapshotCache()`. Make more sense.

**Review by shargon — COMMENTED** [review]
all are changed?

**Comment by Jim8y** [comment]
> all are changed?

what you mean?

File: src/Neo/NeoSystem.cs (hunk 1 above)
Comment (Jim8y) [reply src/Neo/NeoSystem.cs:291]: may or may not, it was get snapshot, so its get snapshot cache. I dont have intention to change it that much.

**Comment by shargon** [comment]
> > all are changed?
> 
> what you mean?

if we don't call to the obsolete methods

**Review by shargon — DISMISSED** [review]

**Review by vncoelho — COMMENTED** [review]
Do we need to leave the obsolete methods in the code?

**Comment by Jim8y** [comment]
> Do we need to leave the obsolete methods in the code?

Yes, cause people could be using them since they are public, we can not remove them until the verion after the next version.

**Review by vncoelho — APPROVED** [review]

**Comment by Jim8y** [comment]
@shargon

**Review by shargon — APPROVED** [review]

**Pull request merged**